### DISCUSSION TITLE
Höhenabfrage - wir müssen es nochmal versuchen 🤔

### DISCUSSION THREAD

**Discussion by nicole-vlbg** [e-netze/webgis-community#125] [Q&A]
Das Thema war eigentlich schon abgeschlossen, wir würden es aber gerne nochmal ins Gespräch bringen, da wir leider nach wie vor immer wieder mit Anfragen konfrontiert werden, die das Anschreiben der Gelände- u. Oberflächenhöhen  direkt in der Karte betreffen, wie das eben bei der Höhenabfrage der Fall war.
Das Höhenabfragetool haben wir zwar inzwischen aus dem webgis5 genommen und verweisen auch auf die Möglichkeit mit dem Ausdruck, aber damit sind unseren Anwender einfach nicht zufrieden.

Anbei die Aktuellste Anmerkung eines Anwenders:

"Hallo! Wir hätten eine Anmerkung zum neuen Atlas: Wir verwenden häufig die Funktionen "Koordinaten / Höhe" oder "Höhenmodell abfragen". Dabei wurden im alten Atlas die Punkte in der Karte beschriftet, was sehr übersichtlich war. Im neuen Atlas ist das nicht mehr der Fall, die Punkte werden nur in einer Liste links angeschrieben, sodass man bei jedem Punkt mit dem Blick von Karte auf Liste wechseln muss. Gibt es die Möglichkeit, das wieder zu implementieren? Vielen Dank für Eure Rückmeldung! Mit freundlichen Grüßen,.."

Vielleicht gibt es doch eine Möglichkeit.

**Comment by gis-eni** [comment]
Das XY Werkzeug bleibt in WebGIS 5 wie es ist. Eine Möglichkeit, dass der Anwender die Punkt in der Karten beschriften kann (nach Koordinaten oder Höhenwerten) kommt nicht. Eventuell kann man das Werkzeug dann in WebGIS 6 erweitern und diese Möglichkeit anbieten.

Für WebGIS 5 kann ich noch folgenden Workaround anbieten:

* Lade die letzte Version herunter (5.23.5001 oder höher)
* Erweitere deine ``custom.js`` um folgenden Eintrag:

``` Javascript
if (webgis.markerIcons["query_result"]["webgis.tools.coordinates"]) {
    webgis.markerIcons["query_result"]["webgis.tools.coordinates"].label = function (i, f) {
        var label = '';
        for (let property in f.properties) {
            if (property == "DGM 10m" || property == "ALS DOM") {
                if (f.properties[property].length > 0 && f.properties[property][0]) {
                    if (label) {
                        label += '\n';
                    }

                    label += `${property}: ${f.properties[property][0]}`;
                }
            }
        }
        return label;
    };
}
```

Hier wird die Defintion der Marker um die Funktion ``label()`` erweitert (die eigentliche Definition der Marker für Koordinaten wird in der ``custom-recommendation.js`` definiert. Die ``custom.js`` wird nachher geladen, und erweitert diese Defintion).

Im Beispiel heißen die beiden Höhen System ``DGM 10m`` und ``ALS DOM``. Diese Namen musst du auf deine Höhensystem anpassen.
Das Ergebnis schaut dann wie folgt aus:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/02f9190e-bda9-46d5-92b9-2ce055b70630)

Wie gesagt, eingestellt wird das von dir, also vom Administrator. Du kannst das in der ``custom.js`` natürlich auf nur für eine Bestimmte Karte einschränken oder für unterschiedliche Karten, andere Höhensystem anzeigen:

``` javascript

if (mapUrlName === "Basiskarte und Kataster") { 

  // ...

}

```
